gateway/blocks_gateway.go

```diff
@@ -223,38 +231,208 @@ func (api *BlocksGateway) Head(ctx context.Context, path ImmutablePath) (Content
 	return md, fileNode, nil
 }
 
-func (api *BlocksGateway) GetCAR(ctx context.Context, path ImmutablePath) (ContentPathMetadata, io.ReadCloser, <-chan error, error) {
-	// Same go-car settings as dag.export command
-	store := dagStore{api: api, ctx: ctx}
+func (api *BlocksGateway) GetCAR(ctx context.Context, p ImmutablePath, params CarParams) (io.ReadCloser, <-chan error, error) {
+	contentPathStr := p.String()
+	if contentPathStr[0:6] != "/ipfs/" {
```

Comment:
```suggestion
	if strings.HasPrefix(contentPathStr, "/ipfs/") {
```
This never risk a panic if we somehow get a path that is too short.

Reply:
Sure, although IIRC this shouldn't happen anyway and even if it did any sane gateway implementation would recover.